activerecord/lib/active_record/nested_attributes.rb

```diff
@@ -355,12 +355,17 @@ def accepts_nested_attributes_for(*attr_names)
         options.update(attr_names.extract_options!)
         options.assert_valid_keys(:allow_destroy, :reject_if, :limit, :update_only)
         options[:reject_if] = REJECT_ALL_BLANK_PROC if options[:reject_if] == :all_blank
+        options[:class] = self
 
         attr_names.each do |association_name|
           if reflection = _reflect_on_association(association_name)
             reflection.autosave = true
             define_autosave_validation_callbacks(reflection)
 
+            if nested_attributes_options.dig(association_name.to_sym, :class) == self
+              raise ArgumentError, "Already declared #{association_name} as an accepts_nested_attributes association for this class."
```

Comment:
We use a model concern across multiple models that add some some association logic as well as a "generic" accepts_nested_attribute_for.

and for some models we override it to add specific options ( `allow_destroy` )

Is there a way to add a config to make raising an argument optional? @rafaelfranca  @joshuay03

Reply:
This is a case of inheritance that I think we should support. But I don't think we should make this configurable. Either we raise when the same class define the same `accepts_nested_attribute_for` more than once, or we don't.

As I believe your concern is using `included` to add the `accepts_nested_attribute_for` both definitions are owned by the same class. This is a valid use case in my opinion. So I think I'll just revert this change.

Reply:
@rafaelfranca

In my opinion, we should be raising despite this use case. I would argue that there's a higher chance of accidentally overriding it this way rather than purposefully doing so, which was the original intent behind the feature, to prevent surprising behaviour. However, I respect your decision to revert this.

What do you think about raising only if the options are different to the previous declaration? So for example:

```ruby
# will not raise

accepts_nested_attribute_for :association
accepts_nested_attribute_for :association, update_only: true 

# will not raise - multiple duplicates

accepts_nested_attribute_for :association
accepts_nested_attribute_for :association, update_only: true 
accepts_nested_attribute_for :association

# second declaration will raise

accepts_nested_attribute_for :association: true
accepts_nested_attribute_for :association: true

# third declaration will raise - multiple duplicates

accepts_nested_attribute_for :association: true
accepts_nested_attribute_for :association, update_only: true 
accepts_nested_attribute_for :association, update_only: true 
```

This will satisfy all use cases. Let me know what you think and I'll get a PR up if it's worth addressing.

Reply:
@rafaelfranca @chaadow the original issue had this example:

```ruby
class Post < ActiveRecord::Base
  accepts_nested_attributes_for :comments, update_only: true
  accepts_nested_attributes_for :comments # I feel like this should raise an error on load
end
```

This is a bit contrived, but I still think it should raise, because the first call is totally skipped. In a much bigger model file it might be less obvious why.

Would you accept a PR that still raises for that, but does *not* raise if the calls are in different places. So the above would raise, but this would not:

```ruby
module Commentable
  accepts_nested_attributes_for :comments, update_only: true
end

class Post < ActiveRecord::Base
  include Commentable
  accepts_nested_attributes_for :comments # Don't raise; we are overriding Commentable.
end
```

Reply:
If you find a way to implement that we can raisae, but I don't think there is a way to do that, because your example is more like:

```
module Commentable
  extend ActiveSupport::Concern

  included do
    # The scope here is Post so it is the same as running inside the class itself. I see no way to differentiate this usage
    # from a direct definition inside the class, because they are actually the same.
    accepts_nested_attributes_for :comments, update_only: true
  end
end

class Post < ActiveRecord::Base
  include Commentable
  accepts_nested_attributes_for :comments # Don't raise; we are overriding Commentable.
end
```